tiledb/sm/filesystem/vfs.h

```diff
@@ -93,11 +127,19 @@ class VFS {
   /*     CONSTRUCTORS & DESTRUCTORS    */
   /* ********************************* */
 
-  /** Constructor. */
-  VFS();
+  /** Constructor.
+   * @param parent_stats The parent stats to inherit from.
+   * @param compute_tp Thread pool for compute-bound tasks.
+   * @param io_tp Thread pool for io-bound tasks.
+   * @param config Configuration parameters.
+   **/
+  VFS(stats::Stats* parent_stats,
+      ThreadPool* compute_tp,
+      ThreadPool* io_tp,
+      const Config& config);
 
   /** Destructor. */
-  ~VFS() = default;
+  ~VFS();
```

Comment:
If the code added to the destructor can be moved to the S3 class, this can go back to being declared `default`.